### ISSUE TITLE
Timeout error object doesn't include .timeout property

### ISSUE THREAD

**Issue by quasipickle** [visionmedia/superagent#1488]
I am getting an error triggered by a timeout, but the error object doesn't have a timeout property.

My code:

```javascript
superagent
    .post(window.location.pathname+'query/')
    .set('Content-Type','application/x-www-form-urlencoded')
    .timeout({
        response: 300
        ,deadline: 10000
    })
    .send({query:$input.value})
    .then(
        Response=>{
            if(Response.text == ''){
                this.replaceContent("I wasn't able to find an answer to that.");
            }
            else{
                this.replaceWith(Response.text);
                $input.value = '';
            }
        },
        Error=>{
            console.log(Object.keys(Error));
            if(Error.timeout){
                this.replaceContent("Sorry for taking so long.  I wasn't able to find an answer.");
            }
        }
    );
```

If I remove the call to `.timeout()`, or set the response timeout to something sensible - I've set it short for testing - then the error function doesn't get called.  Otherwise, it does get called, but Error only has `code`,`status`,`method`,`url` - no `timeout`.

**Comment by unglud** [comment]
same here

**Comment by niftylettuce** [comment]
PR welcome

**Comment by webarnes** [comment]
It's something specific to the promise interface. If you wrap request.end with util.promisify, it will work as expected:

```
const request = superagent.get('/').timeout(300);
const promise = util.promisify(request.end.bind(request));

promise.then(successFn, rejectFn) // Timeout error
request.then(successFn, rejectFn) // Abort error
```

I suspect it is an issue with the timeoutError function passing the error to the callback which only gets set when end() is called.

**Comment by buccfer** [comment]
The issue is here

```
RequestBase.prototype.then = function (resolve, reject) {
  var _this = this;

  if (!this._fullfilledPromise) {
    var self = this;

    if (this._endCalled) {
      console.warn('Warning: superagent request was sent twice, because both .end() and .then() were called. Never call .end() if you use promises');
    }

    this._fullfilledPromise = new Promise(function (resolve, reject) {
      self.on('abort', function () {
        var err = new Error('Aborted');
        err.code = 'ABORTED';
        err.status = _this.status;
        err.method = _this.method;
        err.url = _this.url;
        reject(err);
      });
      self.end(function (err, res) {
        if (err) reject(err);else resolve(res);
      });
    });
  }

  return this._fullfilledPromise.then(resolve, reject);
};
```

The problem is that the `_timeoutError` function calls abort at https://github.com/visionmedia/superagent/blob/master/src/request-base.js#L717 so this is emitting an `'abort'` event before `end` is being called.

**Comment by niftylettuce** [comment]
thanks @a631807682 	for the PR #15140

PR merged and version bumped

released to npm as v5.1.4

```sh
npm install superagent@latest
```

or with yarn:

```sh
yarn add superagent@latest
```
